packages/headless/src/app/headless-engine.ts

```diff
@@ -382,6 +393,10 @@ export class HeadlessEngine<Reducers extends ReducersMapObject>
             search?.preprocessQuerySuggestResponseMiddleware ||
             NoopPostprocessQuerySuggestResponseMiddleware,
         }),
+        caseAssistAPIClient: new CaseAssistAPIClient({
+          logger: this.logger,
+          renewAccessToken: () => this.renewAccessToken(),
+        }),
```

Comment:
Rather than adding the caseAssistAPIClient to all engines, could we add it selectively to just CaseAssist engines?

Otherwise, if headless supports `n` apis, we will be bundling `n` api clients, many of which may not be used.

Another thought, atm we have an `Engine` interface. There might be a need to extend this and have `CaseAssistEngine` so that async thunks have the correct intellisense?

Reply:
> Rather than adding the caseAssistAPIClient to all engines, could we add it selectively to just CaseAssist engines?

It would be nice to be able to have only the relevant API(s) on the engine. At this moment, I don't really see how to dynamically make it available in the `ThunkExtraArguments` though.

That's clearly something we have to figure out sooner than later because we plan on adding another API client for User Actions quite soon.

Reply:
@samisayegh @olamothe I feel this discussion is closely related to another one we had regarding the engine responsibilities (i.e., should a single engine power a UI containing search, recommendations, case assist, etc at once).

Sami, you were proposing that an engine should handle only one use-case at a time. A UI containing many use cases should contain many specialized engines, and the customer should synchronize them together. Olivier, you seem to prefer to have a single engine that connects everything.

Now, since both these discussions are blocking future Case Assist development in headless, I would ask you to:

1. Decide how a UI with multiple use-cases should be handled in headless, and clearly communicate that vision.
2. Implement the missing facilities in headless, so new APIs could be easily added and used in actions.

Meanwhile, I'll remove from this PR the part where `CaseAssistAPIClient` integrates with the engine, and I'll work on designing the future Case Assist actions and controllers.

Once headless engine is ready to receive new API clients, I'll be able to connect the client and implement the related actions.

Does it make sense to you?

Reply:
> Sami, you were proposing that an engine should handle only one use-case at a time. A UI containing many use cases should contain many specialized engines, and the customer should synchronize them together.

That's the current plan on how we wanna move forward. Not a single engine that connects everything.

> Once headless engine is ready to receive new API clients, I'll be able to connect the client and implement the related actions.
> Does it make sense to you?

Yep !

Reply:
Hi Luc, I will do a spike into this today to see what is possible regarding the API clients. As far as reducing the scope of this PR, that makes sense,